### PR TITLE
Time selection for fvcom_tools and unit test

### PR DESCRIPTION
**Pull request by dmwright526**
## DESCRIPTION OF CHANGES: 
Updates have been made to fvcom_tools to allow for the read in of either a cold start surface file from chgres_cube or a warm start surface file generated from a restart of the UFS-SRW. Updates have also been made to allow for the use of fvcom data files with multiple time indices. This requires two additional command line inputs:

- "warm" or "cold" flag. "warm" will allow the tool to read in files generated from a restart. "cold" will read output files from chgres_cube.
- String of time slice to be used in fvcom data file. For example, '2020-01-31T18:00:00.000000'

## TESTS CONDUCTED: 
Tests were conducted on Jet and Hera.

A unit test has also been added, which passes on Jet and Hera. It passes the "Intel" test on Github, but fails the "linux-mac-mpi-nceplibs", "netcdf-versions", and "debug-docs-test_coverage" for unknown reasons.

## DEPENDENCIES:
None

## DOCUMENTATION:
Updates have been made to fvcom_readme.md to include the required input variables.

## ISSUE: 
Fixing issues addressed in: https://github.com/ufs-community/UFS_UTILS/issues/586

## CONTRIBUTORS (optional): 
@JeffBeck-NOAA

### CODE REVIEW

**Comment by LarissaReames** [comment]
I was able to reproduce the error that the Github actions are experiencing by using the gnu compiler and the Cmake debug build type. I took a quick glance at the data type where the error is occurring (fcst_nwp) and there are a bunch of pointers in there, and the error is an invalid memory reference error, so I'm guessing that data type needs to be looked at more closely.

**Comment by JeffBeck-NOAA** [comment]
@GeorgeGayno-NOAA, @dmwright526 has two small NetCDF files that are used for the unit tests in this PR.  Since we don't want to store any binary files in the repo, can we get these added to the EMC FTP server?  If we can, how can David modify his unit test to download those files?  Thanks.

**Comment by GeorgeGayno-NOAA** [comment]
> @GeorgeGayno-NOAA, @dmwright526 has two small NetCDF files that are used for the unit tests in this PR. Since we don't want to store any binary files in the repo, can we get these added to the EMC FTP server? If we can, how can David modify his unit test to download those files? Thanks.

I can ask @KateFriedman-NOAA to host any files for us.

**Comment by GeorgeGayno-NOAA** [comment]
> @GeorgeGayno-NOAA, @dmwright526 has two small NetCDF files that are used for the unit tests in this PR. Since we don't want to store any binary files in the repo, can we get these added to the EMC FTP server? If we can, how can David modify his unit test to download those files? Thanks.

To modify his unit test to download data from the server, he can follow this example:

First, add a 'files.txt' file, such as:
https://github.com/ufs-community/UFS_UTILS/blob/develop/tests/chgres_cube/data/files.txt

Then, invoke the file in the CMakeLists.txt file as follows:
https://github.com/ufs-community/UFS_UTILS/blob/develop/tests/chgres_cube/CMakeLists.txt#L19

**Comment by dmwright526** [comment]
Thanks @GeorgeGayno-NOAA . The files are currently located in the data folder for the test that need to be staged. I will go ahead and work on modify the code to download the data. Once that data is staged I will remove it from my fork.

Thanks @LarissaReames-NOAA for checking out the error. I will do some work on Cheyenne to see if I can fix the problem with the GNU compiler.

**Comment by KateFriedman-NOAA** [comment]
> I can ask @KateFriedman-NOAA to host any files for us.

I'm happy to add files to our static "fix file" sets or push some files to the ftp server. The latter is not necessarily guaranteed to be official or long-term. The copy of the fix files on the ftp server is official and long-term though since we support those files on our NOAA HPC machines and host them on the ftp server for outside users. Do these two binary files qualify as fix-type files? See the fix file sets here on the ftp server for examples of what are contained within:

https://ftp.emc.ncep.noaa.gov/static_files/public/UFS/GFS/

The "fix" folder there is the developmental set. The other two "fix_nco_gfsv##" sets are the frozen fix sets from the GFSv15 and GFSv16 operational systems. I add the new CO2 files that we receive each fall to all sets when they arrive, including the frozen ops ones, but I otherwise leave the frozen sets frozen. Let me know what you need, thanks!

**Comment by GeorgeGayno-NOAA** [comment]
> > I can ask @KateFriedman-NOAA to host any files for us.
> 
> I'm happy to add files to our static "fix file" sets or push some files to the ftp server. The latter is not necessarily guaranteed to be official or long-term. The copy of the fix files on the ftp server is official and long-term though since we support those files on our NOAA HPC machines and host them on the ftp server for outside users. Do these two binary files qualify as fix-type files? See the fix file sets here on the ftp server for examples of what are contained within:
> 
> https://ftp.emc.ncep.noaa.gov/static_files/public/UFS/GFS/
> 
> The "fix" folder there is the developmental set. The other two "fix_nco_gfsv##" sets are the frozen fix sets from the GFSv15 and GFSv16 operational systems. I add the new CO2 files that we receive each fall to all sets when they arrive, including the frozen ops ones, but I otherwise leave the frozen sets frozen. Let me know what you need, thanks!

@KateFriedman-NOAA These files are only used by our Github unit tests. They would go somewhere here:
https://ftp.emc.ncep.noaa.gov/static_files/public/UFS/ufs_utils/unit_tests/  They don't need to be hosted on our supported machines.

**Comment by KateFriedman-NOAA** [comment]
> @KateFriedman-NOAA These files are only used by our Github unit tests. They would go somewhere here: https://ftp.emc.ncep.noaa.gov/static_files/public/UFS/ufs_utils/unit_tests/ They don't need to be hosted on our supported machines.

Okie dokie, please point me to the files for pickup and I'll put them in that location on the ftp server. Thanks!

**Comment by dmwright526** [comment]
Thanks for the help, @KateFriedman-NOAA! The two netcdf files (fvcom_unittest.nc, sfcdata_unittest.nc) can be found here:

https://github.com/dmwright526/UFS_UTILS/tree/feature/TimeSelectingFVCOM/tests/fvcom_tools/data

**Comment by KateFriedman-NOAA** [comment]
@dmwright526 Done, see the files here:
https://ftp.emc.ncep.noaa.gov/static_files/public/UFS/ufs_utils/unit_tests/fvcom_unittest.nc
https://ftp.emc.ncep.noaa.gov/static_files/public/UFS/ufs_utils/unit_tests/sfcdata_unittest.nc

Let me know if you want a subfolder at that level to hold those two files, similar to the other two folders there. Thanks!

**Comment by GeorgeGayno-NOAA** [comment]
> @dmwright526 Done, see the files here: https://ftp.emc.ncep.noaa.gov/static_files/public/UFS/ufs_utils/unit_tests/fvcom_unittest.nc https://ftp.emc.ncep.noaa.gov/static_files/public/UFS/ufs_utils/unit_tests/sfcdata_unittest.nc
> 
> Let me know if you want a subfolder at that level to hold those two files, similar to the other two folders there. Thanks!

@KateFriedman-NOAA Yes, please use sub-folders. The number of unit tests is increasing rapidly.

**Comment by KateFriedman-NOAA** [comment]
> Yes, please use sub-folders. The number of unit tests is increasing rapidly.

Okie dokie...one for each file?
```
/fvcom/fvcom_unittest.nc
/sfcdata/sfcdata_unittest.nc
```

...or together and if together, what folder name would you like?

**Comment by GeorgeGayno-NOAA** [comment]
> > Yes, please use sub-folders. The number of unit tests is increasing rapidly.
> 
> Okie dokie...one for each file?
> 
> ```
> /fvcom/fvcom_unittest.nc
> /sfcdata/sfcdata_unittest.nc
> ```
> 
> ...or together and if together, what folder name would you like?

I would place both files in a subdirectory called `fvcom_tools`.

**Comment by KateFriedman-NOAA** [comment]
> I would place both files in a subdirectory called `fvcom_tools`.

Done, see the new folder with the two files here:
https://ftp.emc.ncep.noaa.gov/static_files/public/UFS/ufs_utils/unit_tests/fvcom_tools/

Let me know if you need anything else related to this, thanks!

**Comment by GeorgeGayno-NOAA** [comment]
> I was able to reproduce the error that the Github actions are experiencing by using the gnu compiler and the Cmake debug build type. I took a quick glance at the data type where the error is occurring (fcst_nwp) and there are a bunch of pointers in there, and the error is an invalid memory reference error, so I'm guessing that data type needs to be looked at more closely.

I was also able to reproduce the error on Hera. I was hoping the debug flags to point to the problem. But no such luck.

@dmwright526 and @JeffBeck-NOAA Do you still want to merge, then fix the GNU test later?

**Comment by JeffBeck-NOAA** [comment]
> > I was able to reproduce the error that the Github actions are experiencing by using the gnu compiler and the Cmake debug build type. I took a quick glance at the data type where the error is occurring (fcst_nwp) and there are a bunch of pointers in there, and the error is an invalid memory reference error, so I'm guessing that data type needs to be looked at more closely.
> 
> I was also able to reproduce the error on Hera. I was hoping the debug flags to point to the problem. But no such luck.
> 
> @dmwright526 and @JeffBeck-NOAA Do you still want to merge, then fix the GNU test later?

@GeorgeGayno-NOAA, due to the time-sensitive nature of this work, yes, we'd like to merge, then address the GNU unit test issue.  The code changes in this PR are only related to the FVCOM tools and will impact only those people running that code (GSL and GLERL).  @dmwright526 will create an issue to follow-up with the GNU unit test failure, and that will be fixed in the near future.

File: sorc/fvcom_tools.fd/fvcom_readme.md [hunk 1]
```diff
@@ -19,16 +19,23 @@
  Installation depends on the netCDF library and cmake.
```
Comment (GeorgeGayno-NOAA) [inline sorc/fvcom_tools.fd/fvcom_readme.md]: We are placing all the 'readme' files in the ./docs directory. So rename this file as 'user_guide.md' and replace the version in ./docs.

**Comment by GeorgeGayno-NOAA** [comment]
@dmwright526 I will merge. But be sure to open an issue to fix the GNU test.

**Review by GeorgeGayno-NOAA — APPROVED** [review]

**Comment by JeffBeck-NOAA** [comment]
Thank you @GeorgeGayno-NOAA.  @dmwright526 has opened an issue for the gnu unit test [here](https://github.com/ufs-community/UFS_UTILS/issues/596).

**Pull request merged**